cpp/src/barretenberg/proof_system/polynomial_store/polynomial_store.test.cpp

```diff
@@ -0,0 +1,114 @@
+#include <cstddef>
+#include <gtest/gtest.h>
+
+#include "polynomial_store.hpp"
+#include "barretenberg/polynomials/polynomial.hpp"
+
+namespace bonk {
+
+using namespace barretenberg;
+using Fr = barretenberg::fr;
+using Polynomial = barretenberg::Polynomial<Fr>;
+
+// Test basic put and get functionality
+TEST(PolynomialStore, PutThenGet)
+{
+    PolynomialStore<Fr> polynomial_store;
+
+    // Instantiate a polynomial with random coefficients
+    Polynomial poly(1024);
+    for (auto& coeff : poly) {
+        coeff = Fr::random_element();
+    }
+
+    // Make a copy for comparison after original is moved into container
+    Polynomial poly_copy(poly);
+
+    // Move the poly into the container
+    polynomial_store.put("id", std::move(poly));
```

Comment:
Is there a need for std::move here? You are doing it inside anyway

Reply:
Since this is just a test I suppose there is no real need, but since most of these tests are meant to be a simple illustration of how the store is to be used, I think it makes sense to demonstrate the correct pattern.